scipy/interpolate/_rgi.py

```diff
@@ -217,20 +217,39 @@ def __init__(self, points, values, method="linear", bounds_error=True,
             self._validate_grid_dimensions(points, method)
         self.method = method
         self.bounds_error = bounds_error
+        values = self._check_values(values)
+        self._check_dimensionality(points, values)
+        self._check_fill_value(values, fill_value)
+        self.fill_value = fill_value
+        self._descending_dimensions = self._check_points(points)
+        self.grid = tuple([np.flip(p) if i in self._descending_dimensions
+                           else np.asarray(p)
+                           for i, p in enumerate(points)])
```

Comment:
gh-16483 needs to make sure that the grid points are floats: `self.grid = tuple([np.asarray(p, dtype=float) for p in points])`, could make it like this here already.

Reply:
I've took the chance to restructure and possibly make this a bite more readable.